File: bundles/org.palladiosimulator.simexp.dsl.ea.launch/src/main/java/org/palladiosimulator/simexp/dsl/ea/launch/pareto/ParetoFront.java
package org.palladiosimulator.simexp.dsl.ea.launch.pareto;

import java.util.List;
import java.util.Map;

public class ParetoFront {
    public static class ParetoEntry {
        public final double fitness;
        public final Map<String, Object> optimizables;

        public ParetoEntry(double fitness, Map<String, Object> optimizables) {
            this.fitness = fitness;
            this.optimizables = optimizables;
        }
    }

    public final List<ParetoEntry> paretoFront;

    public ParetoFront(List<ParetoEntry> entries) {
        this.paretoFront = entries;
    }

}

File: datavisualization/python/simulation_result/pareto/pareto_front.py
from dataclasses import dataclass


@dataclass(frozen=True)
class ParetoEntry:
    id: str
    score: float
    fitness: float
    optimizables: dict[str, str]
    average_energy_consumption: float
    average_packet_loss: float


@dataclass(frozen=True)
class ParetoFront:
    generation: int
    entries: list[ParetoEntry]
